Adding Time alive bar, Raid damage and Raid time alive.

cwhavener opened the pull request:
adding time alive functionality as well as time alive over the course of a raid boss (keeps track of total damage/time alive on all pulls of a boss) will reset on boss kill or zone leave.

also added PKTTriggerStartNotify which can be used to tell if a trigger event (such as boss kill/wipe) have started. Currently being used to tell if a wipe/kill has happened to help get rid of the bloat encounters as PKTRaidBossKillNotify is sent not only on boss kill as previously thought, but on Wipes and boss start as well.

Raid DPS is also based on DPS while alive, not DPS averaged over entire encounter length

shalzuth commented:
@cwhavener what's the point of PlayerEntities?
Where did StartType.cs come from? Is that TriggerSignalType from the SDK?
I'm not understanding the point of RaidInfos - why not just create new encounter for each wipe? (no one really wants to use the old encounter)

cwhavener commented:
PlayerEntities is there because after every wipe/kill I cycle through all players to figure out if they have died to calculate time alive in the encounter. the Entities can get rather large so in the interest of computation time I added Player entities so that it would cycle through 8 max.  StartType.cs came from Herysia, and yes it is TriggerSignalType from the new TriggerSignalStart packet. RaidInfos is a bit specific but is used for cumulative information between pulls on a boss, for example if you wanted to look at total time alive over multiple pulls for gate 3 vykas or look at total damge done for one gate of vykas. This part is a bit specific, but people in my static wanted  it so I added it. can easily be removed if you think its too specific.

shalzuth commented:
I've merged it - but I do worry it carries technical debt.
